### PR TITLE
extra attrs for xrf spectrum dict; file suffix as optional property

### PR DESCRIPTION
**Pull request by meguiraun**
small changes for the abstract XRF

### CODE REVIEW

File: mxcubecore/HardwareObjects/abstract/AbstractXRFSpectrum.py [hunk 1]
```diff
@@ -89,23 +90,28 @@ def start_spectrum(
...
         self.spectrum_info_dict["filename"] = ""
         # Create the data and the archive directory (if needed) and files
         if data_dir:
             if not self.create_directory(data_dir):
                 self.update_state(self.STATES.FAULT)
                 return False
             filename = self.get_filename(data_dir, prefix)
-            self.spectrum_info_dict["filename"] = filename + ".dat"
+            self.spectrum_info_dict["filename"] = filename + "." + self.file_suffix
+            self.spectrum_info_dict["spectrum_directory"] = os.path.dirname(filename)
```
Comment (beteva) [inline mxcubecore/HardwareObjects/abstract/AbstractXRFSpectrum.py:102]: Does prefix and  spectrum_directory exists in the data model?

File: mxcubecore/HardwareObjects/abstract/AbstractXRFSpectrum.py [hunk 2]
```diff
@@ -89,23 +90,28 @@ def start_spectrum(
...
                 return False
             filename = self.get_filename(archive_dir, prefix)
-            self.spectrum_info_dict["scanFileFullPath"] = filename + ".dat"
+            self.spectrum_info_dict["scanFileFullPath"] = (
+                filename + "." + self.file_suffix
+            )
             self.spectrum_info_dict["jpegScanFileFullPath"] = filename + ".png"
             self.spectrum_info_dict["annotatedPymcaXfeSpectrum"] = filename + ".html"
             self.spectrum_info_dict["fittedDataFileFullPath"] = filename + "_peaks.csv"
+            self.spectrum_info_dict["archive_directory"] = archive_dir
```
Comment (beteva) [inline mxcubecore/HardwareObjects/abstract/AbstractXRFSpectrum.py:114]: Does archive_directory exists in the data model?

File: mxcubecore/HardwareObjects/abstract/AbstractXRFSpectrum.py [hunk 3]
```diff
@@ -63,6 +63,7 @@ def __init__(self, name):
     def init(self):
         """Initialisation"""
         self.default_integration_time = self.get_property("default_integration_time", 3)
+        self.file_suffix = self.get_property("file_suffix", "dat")
```
Comment (beteva) [inline mxcubecore/HardwareObjects/abstract/AbstractXRFSpectrum.py:66]: This is better than hardcoding the suffix 👍

**Review by beteva — APPROVED** [review]

File: mxcubecore/HardwareObjects/abstract/AbstractXRFSpectrum.py (hunk 1 above)
Comment (meguiraun) [reply mxcubecore/HardwareObjects/abstract/AbstractXRFSpectrum.py:102]: I am lost. which data model?

**Pull request merged**